nvm.sh

```diff
@@ -695,6 +695,82 @@ nvm_binary_available() {
   nvm_version_greater_than_or_equal_to "$(nvm_strip_iojs_prefix "${1-}")" v0.8.6
 }
 
+nvm_set_colors() {
+  if [ nvm_has_colors ]
+  then
+    COLORS=$1
+    INSTALLED_COLOR="$(nvm_print_color_code ${COLORS:0:1})"
+    LTS_AND_SYSTEM_COLOR="$(nvm_print_color_code ${COLORS:1:1})"
+    CURRENT_COLOR="$(nvm_print_color_code ${COLORS:2:1})"
+    NOT_INSTALLED_COLOR="$(nvm_print_color_code ${COLORS:3:1})"
+    DEFAULT_COLOR="$(nvm_print_color_code ${COLORS:4:1})"
+    printf "Setting colors to: \033[${INSTALLED_COLOR}${COLORS:0:1}\033[${LTS_AND_SYSTEM_COLOR}${COLORS:1:1}\033[${CURRENT_COLOR}${COLORS:2:1}\033[${NOT_INSTALLED_COLOR}${COLORS:3:1}\033[${DEFAULT_COLOR}${COLORS:4:1}\033[0m"
+    export NVM_COLORS="$INSTALLED_COLOR:$LTS_AND_SYSTEM_COLOR:$CURRENT_COLOR:$NOT_INSTALLED_COLOR:$DEFAULT_COLOR"
+  fi
+}
+
+nvm_get_colors() {
+  if [ $NVM_COLORS ]; then
+    INSTALLED_COLOR="$(cut -d':' -f1 <<<$NVM_COLORS)"
+    SYSTEM_COLOR="$(cut -d':' -f2 <<<$NVM_COLORS)"
+    CURRENT_COLOR="$(cut -d':' -f3 <<<$NVM_COLORS)"
+    NOT_INSTALLED_COLOR="$(cut -d':' -f4 <<<$NVM_COLORS)"
+    DEFAULT_COLOR="$(cut -d':' -f5 <<<$NVM_COLORS)"
+    LTS_COLOR="$(echo $SYSTEM_COLOR | tr '0;' '1;')"
+  else 
+    CURRENT_COLOR="0;32m"
+    INSTALLED_COLOR="0;34m"
+    LTS_COLOR="1;33m"
+    NOT_INSTALLED_COLOR="0;31m"
+    DEFAULT_COLOR="0;37m"
+    SYSTEM_COLOR="0;33m"
+  fi
+}
+
+nvm_print_color_code(){
+  case $1 in
+    r) echo "0;31m";;
+    R) echo "1;31m";;
+    g) echo "0;32m";;
+    G) echo "1;32m";;
+    b) echo "0;34m";;
+    B) echo "1;34m";;
+    c) echo "0;36m";;
+    C) echo "1;36m";;
+    m) echo "0;35m";;
+    M) echo "1;35m";;
+    y) echo "0;33m";;
+    Y) echo "1;33m";;
+    k) echo "0;30m";;
+    K) echo "1;30m";;
+    e) echo "0;37m";;
+    W) echo "1;37m";;
+  esac
+}
+
+nvm_format_help_message_colors() {
+  if [ -z "${NVM_NO_COLORS}"  ] && nvm_has_colors
+  then
+    RED_INFO="\033[0;31m r\033[0m/\033[1;31mR\033[0m = \033[0;31mred\033[0m / \033[1;31mbold red\033[0m"
```

Comment:
same here; are these variables meant to be exported? is there perhaps a better way to return information instead of relying on exporting env vars?

Reply:
these still need `local` declarations; also do they need double quotes if there's no var interpolation?

Reply:
Hi @ljharb, with this function, these variables are used in the --help command section further in the file, so they cannot be made local to this function. However, we prefixed the names, as per your suggestion in another function.